libraries/bxdf/usd_preview_surface.mtlx

```diff
@@ -34,21 +50,52 @@
+           doc="Path to the color texture."/>
+    <input name="st" type="vector2" defaultgeomprop="UV0" 
+           doc="Texture coordinate to use to fetch this texture."/>
+    <input name="wrapS" type="string" value="periodic" enum="black,clamp,periodic,mirror" uniform="true" 
+           doc="Wrap mode when reading this texture."/>
+    <input name="wrapT" type="string" value="periodic" enum="black,clamp,periodic,mirror" uniform="true" 
+           doc="Wrap mode when reading this texture."/>           
+    <input name="fallback" type="color4" value="0, 0, 0, 1" 
+           doc="Fallback value used when texture can not be read. Default is ( 0, 0, 0, 1 ) for color texture."/>
+    <input name="scale" type="color4" value="1, 1, 1, 1" uniform="true" 
+           doc="Scale to be applied to all components of the texture. Output is textureValue * scale + bias. Default is (1, 1, 1, 1) for color texture."/>
+    <input name="bias" type="color4" value="0, 0, 0, 0" uniform="true" 
+           doc="Bias to be applied to all components of the texture. Output is textureValue * scale + bias. Default is (0, 0, 0, 0) for color texture."/>
     <output name="r" type="float" />
     <output name="g" type="float" />
     <output name="b" type="float" />
     <output name="a" type="float" />
     <output name="rgb" type="color3" />
   </nodedef>
 
+  <nodedef name="ND_UsdUVTexture_23_NormalMap" node="UsdUvTexture" nodegroup="texture2d" version="2.3" isdefaultversion="true" 
+           doc="USD normal map texture reader version 2.3.">
+    <input name="file" type="filename" uniform="true" 
+           doc="Path to the normal map texture."/>
+    <input name="st" type="vector2" defaultgeomprop="UV0" 
+           doc="Texture coordinate to use to fetch this texture."/>
+    <input name="wrapS" type="string" value="periodic" enum="black,clamp,periodic,mirror" uniform="true" 
+           doc="Wrap mode when reading this texture."/>
+    <input name="wrapT" type="string" value="periodic" enum="black,clamp,periodic,mirror" uniform="true" 
+           doc="Wrap mode when reading this texture."/>
+    <input name="fallback" type="vector3" value="0.5, 0.5, 1" 
+           doc="Fallback value used when texture can not be read. Default is ( 0.5, 0.5, 1.0 ) for normal maps."/>
+    <input name="scale" type="vector3" value="2, 2, 2" uniform="true" 
+           doc="Scale to be applied to all components of the texture. Output is textureValue * scale + bias. Default is (2, 2, 2) for normal maps"/>           
+    <input name="bias" type="vector3" value="-1, -1, -1" uniform="true" 
+           doc="Bias to be applied to all components of the texture. Output is textureValue * scale + bias. Default is (-1, -1, 1) for normal maps"/>
+    <output name="r" type="float" />
+    <output name="g" type="float" />
+    <output name="b" type="float" />
+    <output name="a" type="float"  />
+    <output name="rgb" type="vector3" />
```

Comment:
Since the only distinguishing feature of this UsdUvTexture node is its vector3 output, how is a user to specify whether they want a traditional UsdUvTexture node with a vector3 output, or this scaled and biased version for normal maps?  Does this new node effectively assume that _all_ textures with vector3 outputs are normal maps?

Reply:
I've remove the implicit semantic knowledge that this is a normal map but instead it's just the vector3 variant. The user must specify the correct scale and bias if a normal map is the input.